distributed/tests/test_stress.py

```diff
@@ -284,3 +285,29 @@ async def test_no_delay_during_large_transfer(c, s, w):
     nbytes -= nbytes[0]
     assert nbytes.max() < (x_nbytes * 2) / 1e6
     assert nbytes[-1] < (x_nbytes * 1.2) / 1e6
+
+
+@pytest.mark.slow
+@gen_cluster(client=True, Worker=Nanny, nthreads=[("127.0.0.1", 2)] * 6)
+async def test_chaos_rechunk(c, s, *workers):
+    s.allowed_failures = 10000
+
+    plugin = KillWorker(delay="4 s", mode="graceful")
+
+    await c.register_worker_plugin(plugin, name="kill")
+
+    da = pytest.importorskip("dask.array")
+
+    x = da.random.random((10000, 10000))
+    y = x.rechunk((10000, 20)).rechunk((20, 10000)).sum()
+    z = c.compute(y)
+
+    start = time()
+    while time() < start + 10:
+        if z.status == "error":
+            await z
+        if z.status == "finished":
+            return
+        await asyncio.sleep(0.1)
+
+    await z.cancel()
```

Comment:
What does this actually test? Every 4 seconds, all workers will shut down, losing all accrued progress, and will be start being restarted in chain by the nannies. Over time, timings will drift apart and reach a somewhat random distribution of restarts. It's close to impossible to reach z.status=="finished", as it would take way more than 4s to finish (it takes 13s *without KillWorker* on a speedy laptop). Because of the very high allowed_failures, status should *never* be failed. If the cluster deadlocks e.g. due to illegal transitions, the test won't notice.

Reply:
See https://github.com/dask/distributed/pull/6134